### PR TITLE
Signup: remove the passwordless experiment

### PR DESCRIPTION
**Pull request by enejb**
#### Changes proposed in this Pull Request

Fixes https://github.com/Automattic/wp-calypso/pull/60368 which was reverted in https://github.com/Automattic/wp-calypso/pull/60828

#### Testing instructions
Same as https://github.com/Automattic/wp-calypso/pull/60368

### CODE REVIEW

**Comment by donlair** [comment]
@enejb thanks for pulling this together! I found two issues during testing:

1) On the checkout page after signing up in the mobile flow and not selecting a domain, I still see the "You can change this name at any time...". That should not be included since it was part of a losing test variant. [Reference changes from prior PR](https://github.com/Automattic/wp-calypso/pull/60368/files#diff-7400f16bef0397fda9c7d5cabe2a75da337677000fcbc5897f97293d5b5e4018).

Screenshot:

CleanShot 2022-02-09 at 09 37 36@2x

2) WPCC flows in mobile are broken since they don't currently support the email-only flow. That was addressed in [this commit](https://github.com/Automattic/wp-calypso/pull/60368/commits/4b38d5856f23bc50c14c32e2965793161ca18256) in the failed PR.

**Review by donlair — CHANGES_REQUESTED** [review]
See issues detailed in my prior comment. Thanks again for picking this up @enejb!

**Comment by enejb** [comment]
I tested this pretty well and it worked for me. 

But it would be good to verify that things still work as expected. 

And let me know if you have any more changes that you want to see. 
cc @donlair  and @niranjan-uma-shankar

File: client/blocks/signup-form/index.jsx [hunk 1]
```diff
@@ -1030,12 +1030,7 @@ class SignupForm extends Component {
 			);
 		}
 
-		/*
-			AB Test: passwordlessSignup
-			`<PasswordlessSignupForm />` is for the `onboarding` flow.
-			We are testing whether a passwordless account creation and login improves signup rate in the `onboarding` flow
-		*/
-		if ( this.props.isPasswordlessExperiment ) {
+		if ( this.props.isPasswordless ) {
```
Comment (donlair) [inline client/blocks/signup-form/index.jsx]: The wpcc signup flow is failing for me with the email-only mobile experience. I'm seeing the following screens when trying to go through the akismet signup flow:

CleanShot 2022-02-09 at 09 36 38@2x


CleanShot 2022-02-09 at 20 48 54@2x

Excluding wpcc should fix that:
`if ( this.props.isPasswordless && this.props.flowName !== 'wpcc' ) {`

**Review by donlair — DISMISSED** [review]
Still seeing an issue in the wpcc flow, small change should fix it.

**Comment by donlair** [comment]
@p-jackson do you mind taking a look at this PR with the recent slack convo in mind?

File: client/blocks/signup-form/passwordless.jsx [hunk 2]
```diff
@@ -125,6 +125,7 @@ class PasswordlessSignupForm extends Component {
 
 		this.submitStep( {
 			username,
+			user_id: userId,
```
Comment (p-jackson) [inline client/blocks/signup-form/passwordless.jsx:128]: Ah! Fantastic, this was this missing line that was needed 👍

**Review by p-jackson — COMMENTED** [review]
Looks like you've tracked down the issue.
p1644448146144539/1644013502.153649-slack-CFFF01Q4V

It's been a while since I've looked at the `passwordless` step, which is probably why I forgot about it when reviewing #60330.

I'm not a massive fan of auto-generating usernames based on email addresses for privacy reasons. Although that's already how the passwordless step was working so not something you've changed in this PR.

I won't mark it as approved since I haven't reviewed how the experiment assignment is being made, but I'm happy with this from a signup framework point of view.

File: client/blocks/signup-form/index.jsx (hunk 1 above)
Comment (enejb) [reply client/blocks/signup-form/index.jsx]: Thanks for pointing that out. I think it should be working as expected now.

**Review by niranjan-uma-shankar — APPROVED** [review]
LGTM

File: client/signup/steps/user/index.jsx [hunk 3]
```diff
@@ -495,7 +453,7 @@ export class UserStep extends Component {
 					submitButtonText={ this.submitButtonText() }
 					suggestedUsername={ this.props.suggestedUsername }
 					handleSocialResponse={ this.handleSocialResponse }
-					isPasswordlessExperiment={ this.isPasswordlessExperiment() }
+					isPasswordless={ isMobile() }
 					experimentName={ this.state.experiment }
```
Comment (tyxla) [inline client/signup/steps/user/index.jsx:457]: Note that we're still using the `experiment` local state, here. Should we ditch the `experimentName` prop?

Comment (enejb) [reply client/signup/steps/user/index.jsx:457]: Good catch @tyxla! Will create a PR to clean this up.

Comment (tyxla) [reply client/signup/steps/user/index.jsx:457]: Thanks, @enejb!

**Pull request merged**